types/context.go

```diff
@@ -171,6 +172,12 @@ func (c Context) WithTxBytes(txBytes []byte) Context {
 	return c.withValue(contextKeyTxBytes, txBytes)
 }
 
+func (c Context) CacheContext() (Context, func()) {
+	cms := c.multiStore().CacheMultiStore()
+	cc := c.WithMultiStore(cms)
+	return cc, cms.Write
```

Comment:
Why not just make `Write` a function on the `Context` (and avoid passing around a function pointer)?

Reply:
Then the receiver function of that context can execute `Write`. Also, it is not obvious what `Write` should do if a context is not generated by `CacheContext`.